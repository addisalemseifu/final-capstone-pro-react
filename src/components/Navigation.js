import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { getLocation } from '../redux/location/locationSlice';

export default function Navigation() {
  const dispatch = useDispatch();
  const [searchval, setSearchval] = useState('London');
  function searchHandler(searchval) {
    dispatch(getLocation(searchval));
    setSearchval('');
  }
  return (
    <div className="nav">
      <button
        type="button"
        onClick={(e) => {
          e.preventDefault();
          searchHandler(searchval);
        }}
      >
        Search
      </button>
      <input
        type="text"
        placeholder="search for location"
        onChange={(e) => {
          setSearchval(e.target.value);
        }}
      />

    </div>
  );
}
