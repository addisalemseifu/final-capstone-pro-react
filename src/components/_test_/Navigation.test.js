import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Navigation from '../Navigation';
import { renderWithProviders } from '../utils/utils-for-tests';

describe('Navigation Component', () => {
  it('renders correctly without unexpected changes', () => {
    const navigation = renderer.create;

    expect(navigation).toMatchSnapshot();
  });

  it('renders the search button', async () => {
    renderWithProviders(<Navigation />);
    const searchBtn = screen.getByRole('button');
    expect(searchBtn).toBeInTheDocument();
  });

  it('renders the input bar', async () => {
    renderWithProviders(<Navigation />);
    const inptuBar = screen.getByPlaceholderText(/search for location/i);
    expect(inptuBar).toBeInTheDocument();
  });

  it('should be able to type in input', async () => {
    renderWithProviders(<Navigation />);
    const inptuBar = screen.getByPlaceholderText(/search for location/i);
    fireEvent.change(inptuBar, { target: { value: 'USA' } });
    expect(inptuBar.value).toBe('USA');
  });

  it('should have empty input when add button is clicked', async () => {
    renderWithProviders(<Navigation />);
    const inptuBar = await screen.findByPlaceholderText(/search for location/i);
    const searchBtn = await screen.getByRole('button');
    fireEvent.change(inptuBar, { target: { value: 'USA' } });
    fireEvent.click(searchBtn);
    expect(inptuBar.value).toBe('USA');
  });
});
