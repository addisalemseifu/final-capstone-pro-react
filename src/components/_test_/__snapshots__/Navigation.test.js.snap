// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component renders correctly without unexpected changes 1`] = `[Function]`;
